src/evaluators/ConditionalExpression.js

```diff
@@ -36,7 +37,40 @@ export default function(
   }
   invariant(exprValue instanceof AbstractValue);
 
-  if (!exprValue.mightNotBeTrue()) return env.evaluate(ast.consequent, strictCode);
-  if (!exprValue.mightNotBeFalse()) return env.evaluate(ast.alternate, strictCode);
-  return evaluateWithAbstractConditional(exprValue, ast.consequent, ast.alternate, strictCode, env, realm);
+  const consequent = ast.consequent;
+  const alternate = ast.alternate;
+  if (!exprValue.mightNotBeTrue()) return env.evaluate(consequent, strictCode);
+  if (!exprValue.mightNotBeFalse()) return env.evaluate(alternate, strictCode);
+  return AbstractValue.evaluateWithAbstractConditional(
+    realm,
+    exprValue,
+    () => {
+      return realm.evaluateNodeForEffects(consequent, strictCode, env);
+    },
+    () => {
+      let completion = env.evaluateCompletion(alternate, strictCode);
+      if (completion instanceof AbruptCompletion) {
+        throw completion;
+      }
+      if (completion instanceof Reference) {
+        return Environment.GetValue(realm, completion);
+      }
+      invariant(completion instanceof Value);
+      return completion;
+    },
+    () => {
+      return alternate ? realm.evaluateNodeForEffects(alternate, strictCode, env) : construct_empty_effects(realm);
+    },
+    () => {
+      let completion = env.evaluate(consequent, strictCode);
```

Comment:
I think you mean evaluateCompletion here, since evaluate will never return an AbruptCompletion.

Reply:
This code is lifted from `IfStatement`. Is it wrong too?

Reply:
Seems like it. Clearly we need more test coverage in this area.

Reply:
On the other hand, conditional expressions are expressions, so perhaps evaluate is the right thing to do here in both cases and we do not need the check for an AbruptCompletion.